src/oidc-keychain/oidc-keychain

```diff
@@ -32,75 +33,50 @@ help()
   )>&2
 }
 
-AGENT_SHORT="cdg"
-AGENT_SHORT_PARAM="t:"
-SHORT_OPTS="?hVk$AGENT_SHORT$AGENT_SHORT_PARAM"
-AGENT_LONG="confirm,no-autoload,no-scheme,no-webserver,seccomp,console,debug"
-# single colon means required param, double colon means optional param
-AGENT_LONG_PARAM="lifetime:,pw-store::,with-group::"
-LONG_OPTS="help,usage,version,kill,$AGENT_LONG,$AGENT_LONG_PARAM"
-
-RE_AGENT_SHORT="^-(`echo "$AGENT_SHORT"|sed 's/./&|/g'`)$"
-RE_AGENT_SHORT_PARAM="^-(`echo "$AGENT_SHORT_PARAM"|sed 's/://g;s/./&|/g'`)$"
-RE_AGENT_LONG="^--(`echo "$AGENT_LONG"|sed 's/,/|/g'`)$"
-RE_AGENT_LONG_PARAM="^--(`echo "$AGENT_LONG_PARAM"|sed 's/://g;s/,/|/g'`)$"
-
-OPTS="`getopt -o "$SHORT_OPTS" --long "$LONG_OPTS" -n $ME -- "$@"`"
-if [ $? -ne 0 ]; then
-  echo "Try \`$ME --help' or \`$ME --usage' for more information." >&2
-  exit 64
-fi
-eval set -- "$OPTS"
-
 KILL=false
 AGENT_OPTS=""
-while true; do
-  # Look first for options to pass on to oidc-agent
-  if [[ "$1" =~ $RE_AGENT_SHORT ]]; then
-    AGENT_OPTS="$AGENT_OPTS $1"
-  elif [[ "$1" =~ $RE_AGENT_SHORT_PARAM ]]; then
-    AGENT_OPTS="$AGENT_OPTS $1 $2"
-    shift
-  elif [[ "$1" =~ $RE_AGENT_LONG ]]; then
-    AGENT_OPTS="$AGENT_OPTS $1"
-  elif [[ "$1" =~ $RE_AGENT_LONG_PARAM ]]; then
-    if [ -z "$2" ]; then
-      AGENT_OPTS="$AGENT_OPTS $1"
-    else
-      AGENT_OPTS="$AGENT_OPTS $1=$2"
-    fi
-    shift
-  else
-    # Look for own options
-    case "$1" in
-      -k|--kill)
-        KILL=true
-        ;;
-      "-?"|-h|--help)
-        usage
-        help
-        exit
-        ;;
-      --usage)
-        usage
-        exit
-        ;;
-      -V|--version)
-        echo `oidc-agent -V`|sed 's/agent/keychain/'
-        ;;
-      --)
-        shift
-        break
-        ;;
-      *)
+while [ -n $1 ]; do
```

Comment:
This loop doesn't support letting the single hyphen options being passed together.  That's fine for the particular options supported because it doesn't really make sense for them to go together.  However the usage implies that they can be together, so I think just changing the usage statement will do.

Reply:
Well, actually it does. If multiple options are passed together to the script, e.g. `-cg` the whole optionstring is passed to the agent, again as multiple options together `-cg`. Then the agent handles the parsing of the options.
However, there was another problem so I changed this to using `[ -n "$1" ]`

Reply:
I meant only the oidc-keychain options, listed in the usage as [-?Vk].  You're right that oidc-agent options are still handled OK.